bigquery/snippets/snippet.go

```diff
@@ -341,48 +355,99 @@ func basicQuery(client *bigquery.Client, datasetID, tableID string) error {
-	q.QueryConfig.Dst = destRef
-
-	// Run job, then wait until asyncronous execution is complete.
+	// [START bigquery_query_batch]
+	// Build an aggregate table
+	q := client.Query(`
+		SELECT
+  			corpus,
+  			SUM(word_count) as total_words,
+  			COUNT(1) as unique_words
+		FROM ` + "`bigquery-public-data.samples.shakespeare`" + `
+		GROUP BY corpus;`)
+	q.Priority = bigquery.BatchPriority
+	q.QueryConfig.Dst = client.Dataset(dstDatasetID).Table(dstTableID)
+
+	// Run job, then wait until asynchronous execution is complete.
 	job, err := q.Run(ctx)
 	if err != nil {
 		return err
 	}
-	status, err := job.Wait(ctx)
+	// Job is started, and will progress without interaction.
+	// To demonstrate other work being done, we simply sleep here.
+	time.Sleep(5 * time.Second)
+	status, err := job.Status(ctx)
 	if err != nil {
 		return err
 	}
-	if err := status.Err(); err != nil {
+
+	state := "Unknown"
+	switch status.State {
+	case bigquery.Pending:
+		state = "Pending"
+	case bigquery.Running:
+		state = "Running"
+	case bigquery.Done:
+		state = "Done"
+	}
+	// You can continue to monitor job progress until it reaches
+	// the Done state by polling periodically.  In this example,
+	// we simply print the latest status.
+	fmt.Printf("Job %s in Location %s currently in state: %s\n", job.ID(), job.Location(), state)
+
+	// [END bigquery_query_batch]
+	job.Cancel(ctx)
+	return nil
+}
+
+func queryDryRun(client *bigquery.Client) error {
+	ctx := context.Background()
+	// [START bigquery_query_dry_run]
+
+	q := client.Query(
+		`
+		SELECT 
+		   name,
+		   COUNT(*) as name_count
+		FROM ` + "`bigquery-public-data.usa_names.usa_1910_2013`" + `
+		WHERE state = 'WA' 
+		GROUP BY name
+		`)
+	q.DryRun = true
+	// Location must match that of the dataset(s) referenced in the query.
+	q.Location = "US"
+
+	job, err := q.Run(ctx)
+	if err != nil {
 		return err
 	}
-	// At this point, the query has completed and results are persisted to the
-	// destination table.  You can also choose to read from the table.
-	it, err := job.Read(ctx)
-	for {
-		var row []bigquery.Value
-		err := it.Next(&row)
-		if err == iterator.Done {
-			break
-		}
-		if err != nil {
-			return err
-		}
-		fmt.Println(row)
+	//Dry run is not asynchronous, so get the latest status and statistics.
+	status := job.LastStatus()
+	if err != nil {
+		return err
 	}
-	// [END bigquery_query_destination_table]
+	fmt.Printf("This query will process %d bytes\n", status.Statistics.TotalBytesProcessed)
+
+	// [END bigquery_query_dry_run]
 	return nil
 }
 
+func queryWithDestination(client *bigquery.Client, destDatasetID, destTableID string) error {
+	ctx := context.Background()
+	// [START bigquery_query_destination_table]
+	destRef := client.Dataset(destDatasetID).Table(destTableID)
+	q := client.Query("SELECT 17 as my_col")
+	q.Location = "US" // Location must match the dataset(s) referenced in query.
+	q.QueryConfig.Dst = destRef
```

Comment:
Any reason you created a `destRef` var instead of inlining as above in the batch sample? I'd rather drop the unnecessary variable.

Reply:
inlined this one as well.  I originally had more exposition, but dropped it.